### PR TITLE
APP-925 - Fixed wrong parsing for special characters

### PR DESCRIPTION
**Pull request by luanapp**

### CODE REVIEW

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java [hunk 1]
```diff
@@ -26,15 +28,21 @@
...
 import static org.symphonyoss.integration.webhook.jira.JiraParserConstants.LINK_ENTITY_FIELD;
+import static org.symphonyoss.integration.webhook.jira.JiraParserConstants.VISIBILITY_PATH;
+import static org.symphonyoss.integration.webhook.jira.parser.v1.CommentJiraParser.MENTION_MARKUP;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import org.symphonyoss.integration.entity.Entity;
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: Please remove unused imports.

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/JiraMetadataParser.java [hunk 2]
```diff
@@ -121,6 +125,21 @@ protected void preProcessInputData(JsonNode input) {
...
+   * This method parses the issue description to avoid invalid characters
+   * @param input
+   */
+  private void processSummary(JsonNode input) {
+    JsonNode fieldsNode = input.path(ISSUE_PATH).path(FIELDS_PATH);
+    JsonNode summaryNode = fieldsNode.path(SUMMARY_PATH);
+
+    if (summaryNode != null) {
+      SafeString summary = new SafeString(summaryNode.asText());
+      summary.replaceLineBreaks();
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/JiraMetadataParser.java]: You might use: SafeString summary = ParserUtils.escapeAndAddLineBreaks(summaryNode.asText());

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java [hunk 3]
```diff
@@ -182,6 +177,36 @@ private ObjectNode getCommentNode(JsonNode input) {
...
+//    Map<String, User> usersToMention = determineUserMentions(comment);
+//    if (usersToMention != null && !usersToMention.isEmpty()) {
+//      for (Map.Entry<String, User> userToMention : usersToMention.entrySet()) {
+//        User user = userToMention.getValue();
+//
+//        safeComment.safeReplace(new SafeString(userToMention.getKey()),
+//            ParserUtils.presentationFormat(MENTION_MARKUP, user.getUsername()));
+//      }
+//    }
+    safeComment.replaceLineBreaks();
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: You might use: SafeString safeComment = ParserUtils.escapeAndAddLineBreaks(comment);

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java [hunk 4]
```diff
@@ -197,4 +222,19 @@ private ObjectNode getCommentNode(JsonNode input) {
...
+   * JIRA comments may be restricted to certain user groups on JIRA. This is indicated by the
+   * presence of a "visibility"
+   * attribute on the comment. This method will deem a comment as restricted if the "visibility"
+   * attribute is present,
+   * regardless of its content, as it is not possible to evaluate the visibility restriction on JIRA
+   * against the rooms
+   * the webhook will post to.
+   * @param node JIRA payload.
+   * @return Indication on whether the comment is restricted or not.
+   */
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: Fix the javadoc format and also include unit tests to check this flow.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]
Improve unit tests

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java (hunk 1 above)
Comment (luanapp) [reply src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: Fix'd

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java (hunk 3 above)
Comment (luanapp) [reply src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: Fix'd

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java (hunk 4 above)
Comment (luanapp) [reply src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/CommentMetadataParser.java]: Fix'd

File: src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/JiraMetadataParser.java (hunk 2 above)
Comment (luanapp) [reply src/main/java/org/symphonyoss/integration/webhook/jira/parser/v2/JiraMetadataParser.java]: Fix'd

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**